src/adapters/SwivelAdapter.sol

```diff
@@ -122,6 +125,14 @@ contract SwivelAdapter is IAdapter {
             }
         }
 
+        // receive the underlying funds from the user
+        Safe.transferFrom(
+            IERC20(underlying),
+            msg.sender,
+            address(this),
+            total + fee
```

Comment:
We dont normally implement the fee like this IIRC? 

The user commits to spending amount `a`, and with this they are spending `a` + `fee`?

I believe need to be removing the `fee` amount from the final order filled, rather than requiring the user to spend the additional `fee`?

(thats how i remember the previous implementation at least)

Reply:
> I believe need to be removing the fee amount from the final order filled, rather than requiring the user to spend the additional fee?

That's what we have here - `amounts[i] = amounts[i] - fee;` reduces the size of the last order to accommodate the fee. `total` is the amount that is being spent through Swivel, `fee` is the fee paid to Illuminate. `total + fee` is the amount needed to execute the transaction